File: unificacion_maga/fix_vue.php
<?php
$dir = __DIR__ . '/Frontend/src/views/admin/visan';

$replacements = [
    "import axios from 'axios'" => "import api from '@/services/api'",
    "axios.get(import.meta.env.VITE_API_BASE + '/index.php'" => "api.get('/visan/dashboard'",
    "axios.get(import.meta.env.VITE_API_BASE + '/tabla_datos.php'" => "api.get('/visan/tabla'",
    "axios.get(import.meta.env.VITE_API_BASE + '/historial_cambios.php'" => "api.get('/visan/historial'",
    "axios.post(import.meta.env.VITE_API_BASE + '/editar_datos.php'" => "api.post('/visan/editar'",
    "axios.get(import.meta.env.VITE_API_BASE + '/editar_datos.php'" => "api.get('/visan/editar'",
    "axios.post(import.meta.env.VITE_API_BASE + '/importar_excel.php'" => "api.post('/visan/importar'"
];

$files = glob($dir . '/*.vue');
foreach ($files as $file) {
    $content = file_get_contents($file);
    foreach ($replacements as $old => $new) {
        $content = str_replace($old, $new, $content);
    }
    file_put_contents($file, $content);
    echo "Fixed " . basename($file) . "\n";
}

File: unificacion_maga/fix_vue.py
import os
import re

VUE_DIR = r"C:\xampp\htdocs\unificacion_maga\Frontend\src\views\admin\visan"

replacements = [
    (r"import axios from 'axios'", "import api from '@/services/api'"),
    (r"axios\.get\(import\.meta\.env\.VITE_API_BASE \+ '/index\.php'", "api.get('/visan/dashboard'"),
    (r"axios\.get\(import\.meta\.env\.VITE_API_BASE \+ '/tabla_datos\.php'", "api.get('/visan/tabla'"),
    (r"axios\.get\(import\.meta\.env\.VITE_API_BASE \+ '/historial_cambios\.php'", "api.get('/visan/historial'"),
    (r"axios\.post\(import\.meta\.env\.VITE_API_BASE \+ '/editar_datos\.php'", "api.post('/visan/editar'"),
    (r"axios\.get\(import\.meta\.env\.VITE_API_BASE \+ '/editar_datos\.php'", "api.get('/visan/editar'"),
    (r"axios\.post\(import\.meta\.env\.VITE_API_BASE \+ '/importar_excel\.php'", "api.post('/visan/importar'")
]

for filename in os.listdir(VUE_DIR):
    if filename.endswith(".vue"):
        path = os.path.join(VUE_DIR, filename)
        with open(path, "r", encoding="utf-8") as f:
            content = f.read()
        
        for old, new in replacements:
            content = re.sub(old, new, content)
            
        with open(path, "w", encoding="utf-8") as f:
            f.write(content)
        print(f"Fixed {filename}")
